packages/core/src/renderer/getStyleSheetForBucket.ts

```diff
@@ -37,11 +39,21 @@ export function getStyleSheetForBucket(
   target: Document | undefined,
   renderer: GriffelRenderer,
   elementAttributes: Record<string, string> = {},
+  metadata?: Record<string, unknown>,
 ): IsomorphicStyleSheet {
-  if (!renderer.stylesheets[bucketName]) {
+  let stylesheetKey: StyleBucketName | string = bucketName;
+  if (bucketName === 'm' && metadata) {
+    stylesheetKey = metadata['m'] as string;
```

Comment:
```suggestion
    stylesheetKey = bucketName + metadata['m'] as string;
```

Can we use `bucketName` as prefix rather than replace it completely?